src/routes/flush/pipes/invalidation-pattern.dto.validation.pipe.ts

```diff
@@ -0,0 +1,39 @@
+import { HttpStatus, Injectable, PipeTransform } from '@nestjs/common';
+import { ValidateFunction } from 'ajv';
+import { GenericValidator } from '../../../validation/providers/generic.validator';
+import { JsonSchemaService } from '../../../validation/providers/json-schema.service';
+import { InvalidationPatternDto } from '../entities/invalidation-pattern.dto.entity';
+import {
+  invalidationPatternDetailSchema,
+  invalidationPatternDtoSchema,
+} from '../entities/schemas/invalidation-pattern.dto.schema';
+
+@Injectable()
+export class InvalidationPatternDtoValidationPipe
+  implements PipeTransform<any, InvalidationPatternDto>
+{
+  private readonly isValid: ValidateFunction<InvalidationPatternDto>;
+
+  constructor(
+    private readonly genericValidator: GenericValidator,
+    private readonly jsonSchemaService: JsonSchemaService,
+  ) {
+    this.jsonSchemaService.getSchema(
+      'https://safe-client.safe.global/schemas/flush/invalidation-pattern-detail.json',
+      invalidationPatternDetailSchema,
+    );
+    this.isValid = this.jsonSchemaService.getSchema(
+      'https://safe-client.safe.global/schemas/delegates/invalidation-pattern.dto.json',
+      invalidationPatternDtoSchema,
+    );
+  }
+  transform(data: any): InvalidationPatternDto {
+    try {
+      data.invalidate = data.invalidate?.toUpperCase();
```

Comment:
Why do we change it to upper case?

Reply:
To cover the case where the client sends an all-lowercase string in the payload, like `{ "invalidate": "chains" }`. Maybe this could fit better in the repository, where the actual matching against a `InvalidationTarget` happens, what do you think?

Reply:
But the current CGW doesn't support upper case types or?

Asking this because I'm also considering the value that the mapping in the enum provides. For example maybe we can just use:

```typescript
export enum InvalidationTarget {
  Chains,
  Contracts,
  Tokens,
}
```

And as an example:

```typescript
console.log(InvalidationTarget[InvalidationTarget.Chains]) // prints 'Chains'
```

Reply:
My idea was to make this `InvalidationTarget` case insensitive, but I tested the current CGW and it is case sensitive (i.e. it requires a capitalized word like `Contracts`, `Tokens`...) so maybe it makes sense to remove the `toUpperCase()` hack and just use the enum values as you commented 👍🏻

Reply:
Changed in [c8b41c8](https://github.com/5afe/safe-client-gateway-nest/pull/288/commits/c8b41c8d1c87bd4e25d55da26973411044cfafdd)

Reply:
> My idea was to make this `InvalidationTarget` case insensitive, but I tested the current CGW and it is case sensitive (i.e. it requires a capitalized word like `Contracts`, `Tokens`...) so maybe it makes sense to remove the `toUpperCase()` hack and just use the enum values as you commented 👍🏻

For now let's remove it but the `flush` endpoint is mostly used internally so there are no strict requirements to match the previous API (i.e. this is not used by the Safe Transaction Service).

For this one I don't think we should spend a lot of time reworking it but it's something that we can think about in the future. Maybe we end up not needing the endpoint at all in the future (e.g. by having an authenticated redis web interface to perform these type of tasks).